pyartcd/pyartcd/pipelines/ocp4_scan.py

```diff
@@ -136,14 +138,19 @@ async def ocp4_scan(runtime: Runtime, version: str):
     # Create a Lock manager instance
     lock = Lock.BUILD
     lock_manager = locks.LockManager.from_lock(lock)
+
+    # Get lock name and identifier
     lock_name = lock.value.format(version=version)
+    lock_identifier = jenkins.get_build_path()
+    if not lock_identifier:
+        runtime.logger.warning('Env var BUILD_URL has not been defined: a random identifier will be used for the locks')
 
     try:
         # Skip the build if already locked
         if await lock_manager.is_locked(lock_name):
             runtime.logger.info('Looks like there is another build ongoing for %s -- skipping for this run', version)
         else:
-            async with await lock_manager.lock(lock_name):
+            async with await lock_manager.lock(resource=lock_name, lock_identifier=lock_identifier):
```

Comment:
I'm thinking about a pattern where to request a lock, we enforce providing a job path identifier. And maybe we can roll the lock manager creation in there as well

So something like

```
async def request_lock(lock_type, lock_name, lock_identifier, **kwargs):
    lock_manager = locks.LockManager.from_lock(lock_type)
    if not lock_identifier:
        raise ValueError('...')
    return lock_manager.lock(resource=lock_name, lock_identifier=lock_identifier)
```

Reply:
Sounds good to me, but I'd rather do that in a separate PR

Reply:
Looking at it more closely: we need a `LockManager` instance available outside this function to clear redis connection (which happens by calling `lock_manager.destroy()`. What we could do is enforce a `lock_identifier` to be specified as keyword arg [here](https://github.com/openshift-eng/art-tools/blob/main/pyartcd/pyartcd/locks.py#L107). Not sure though if we want to give up the chance of getting an auto-generated random identifier, in cases where we might not be interested in the job that created it